### ISSUE TITLE
Abandoned App

### ISSUE THREAD

**Issue by DarthPleurotus** [moezbhatti/qksms#1881]
Just doing my civic duty and reminding everyone that you should no longer trust this app with your text messages. It isn't maintained, breaks with new updates, and will possibly lose messages that are sent to you with (as far as I can tell) no recourse for recovery.

I guess the developer already had his fun showing off his fancy app and didn't care enough to properly advertise the fact that he has stopped supporting it.

Shame on me for trusting this individual, and shame on him for his indifference.

**Comment by Commenter25** [comment]
If I do community service around your neighborhood, for free, every week, purely because I decided to; and then one day I move and I'm not doing community service anymore... are you going to get pissed at me for not doing the community service anymore?

**Comment by DarthPleurotus** [comment]
Was there a large community of neighbors who had come to trust and depend on your service for important parts of their life? Did you stop stop suddenly, **but leave up all the signs you put up to encourage people to use and depend on the service?** Then yes, definitely. You could say shame on the neighbors for expecting the service and growing reliant, and I do feel foolish for trusting that a stranger wouldn't pack up shop without any warning. But there is a share of blame to be placed on the developer for failing to announce the change.

If people are reliant on a service you provide and you stop providing it, it may leave them worse off than if you had not started providing the service to begin with. Despite being happy with QKSMS when it was functional, I do feel that (in hindsight) I would have been better off never using it to begin with if that meant not suddenly failing to receive messages without warning.

I reread my above comment, and I do regret the tone I used. It was hurtful and had a lot to do with my own emotional state at the time. The title in particular is inappropriate and unbecoming of a collaborative community. The developer deserves an apology and if I hear back from them, I will offer one directly. If not, if you read this, @moezbhatti, I am sorry.

The developer is moving on? Fine. They owe users nothing and are free to proceed with whatever projects they deem to be worthwhile. Developers archive and abandon projects all the time, and open source users understand that. But have the consideration to announce the change, or at least take down the language actively encouraging users. This level of reliability is appropriate for a hobby project, not something promoted as a reliable replacement for the stock SMS app.

**Comment by DarthPleurotus** [comment]
I have edited the title of the post. While I feel that, in the spirit of transparency, records of what is written here shouldn't be hidden or erased (thus I am noting here that the original title was "Dead, Garbage App"), my choice of title was not inappropriate for setting the sort of civil tone that's appropriate for an issues forum.

**Comment by kunago** [comment]
> Was there a large community of neighbors who had come to trust and depend on your service for important parts of their life? Did you stop stop suddenly, **but leave up all the signs you put up to encourage people to use and depend on the service?** Then yes, definitely. You could say shame on the neighbors for expecting the service and growing reliant, and I do feel foolish for trusting that a stranger wouldn't pack up shop without any warning. But there is a share of blame to be placed on the developer for failing to announce the change.

Have you looked at the developer's activity on GitHub? Do you believe there is everything okay when a person suddenly stops developing out of the blue given his activity history? Usually developers don't stop developing this way; they move onto something else, some other projects, but not like this. But I have no further information, it's only an assumption.

> If people are reliant on a service you provide and you stop providing it, it may leave them worse off than if you had not started providing the service to begin with. Despite being happy with QKSMS when it was functional, I do feel that (in hindsight) I would have been better off never using it to begin with if that meant not suddenly failing to receive messages without warning.

This individual gave something to everyone for free. I believe there is not much to say to that. If you can support him financially then probably you will have a well-functioning app and he will very likely make it work the way you want. If not, you cannot blame him (or anyone else) no matter how you feel about it.

**Comment by Commenter25** [comment]
@DarthPleurotus Respect to you for acknowledging the hostility of your tone. Have to admit, didn't expect a response at all. I agree that if halting development is planned and premeditated, it should be announced, so people can fork or seek alternatives.
However, there could be any number of reasons. The developer has a heavy contribution history inside private repositories. My guess is, it's related to their job as a software engineer for Bite, and that took priority.

[They seem to still want to work on QKSMS, and don't have the time and opportunities.](https://www.reddit.com/r/QKSMS/comments/p11gez/is_this_app_still_maintained/h8jhgjb/) There's a sudden drop-off in March 2021, with few commits after, yet the private history stays strong. Maybe there's a big company project, and they simply don't have the time or energy to develop once their shift is over. Maybe they believe QKSMS doesn't need updates at the moment anyway, and adding anything new can wait until private work chills out? I've personally had no issues with the app, however infrequently I use SMS. (Using a Pixel 5, on the latest Android 12 GrapheneOS)

Even with that, there's about a 50 contrib average in their private history, which dropped to 9 in June, and none this month. Even their reddit history has severely dwindled over time. Their online presence is limited, from what I can see. There's not much to go off of, and maybe they intend it to be that way. Their private life is not my concern, and I simply hope they are safe. Perhaps they should've announced a hiatus, perhaps they didn't want to thinking "maybe i'll get to work on it tomorrow, or tomorrow, or tomorrow". Dunno, really. But I do believe the situation is more nuanced than the surface may suggest.

**Comment by moezbhatti** [comment]
Hey everyone, I just wanted to apologize for the lack of updates, and more importantly the lack of communication here.

I know that it's been frustrating to see all progress come to a halt, and I want to make it clear that I share that frustration. I've been constantly thinking about the app and wanting to work on it, but due to a million different reasons in my personal life and with work, I've had a difficult time translating that intention into any meaningful progress.

Part of the challenge has been falling significantly behind the list of requirements for publishing an update to Google Play, which includes adding support for the latest Android versions and making sure the app doesn't break with all of the recent changes to permissions and security. This meant that even when I did have enough time to work on a small feature or bugfix, I couldn't actually do anything about it because I was completely blocked on publishing updates to Google Play – and there was a mountain of work to be done in order to fix that.

Well, over the last few weeks I have climbed that mountain. QKSMS now has full support for Android 13, and satisfies all of the requirements for publishing updates again. I just published an update [v3.10.0](https://github.com/moezbhatti/qksms/releases/tag/v3.10.0) which includes all of those changes, plus a couple small improvements and bug fixes.

None of this is an excuse for the lack of communication. For that, all I can say is that I'm sorry and I'll try to do better.

Moving forward, it'll be much easier for me to work on the app and push out more regular updates. And that's exactly what I intend to do! This will start with focusing on a number of performance improvements and bug fixes before starting to spend time on bigger features & design changes again.

That aside, I want to acknowledge how much it meant to me to see all of the support in this (and other) threads. I beat myself up a bit when I see the growing list of issues in this repo that I haven't responded to, or when I'm reminded of how long ago the "Last updated" date is, or whenever I experience some frustrating thing in the app that I know other people are dealing with too. I see a lot of emails and reviews come in from users who are extremely upset with me for abandoning the app, and I think they're completely in the right to feel that way. I've got no excuse. I let them down, and I hold myself fully accountable. But when I came across these comments and saw strangers on the internet cutting me some slack, it was extremely unexpected and honestly pretty heartwarming. It also showed me that I could maybe cut myself a bit of slack too. This helped relieve a lot of the pressure and stress I had associated with starting work on the app again, and I was able to just get started and do it at a pace that I can manage. So, thank you all for that.

---

@DarthPleurotus I really admire how you handled this. Your ability to reflect and completely change your tone isn't something you see often. Thank you for doing that, and also for leaving the entire conversation history here as reference.

@kunago @Commenter25 thank you for your concern! Happy to say I'm all okay. The severe drop-off in commits was largely a result of how I decided to spend my summer last year, [riding my bike from Canada to Mexico](https://trackleaders.com/tourdivide22i.php?name=Moez_Bhatti).
